test/katex-spec.js

```diff
@@ -872,7 +871,40 @@ describe("A text parser", function() {
     });
 
     it("should parse math within text group", function() {
-        expect(textWithEmbeddedMath).toParse();
+        expect("\\text{graph: $y = mx + b$}").toParse();
+        expect("\\text{graph: \\(y = mx + b\\)}").toParse();
+    });
+
+    it("should parse math within text within math within text", function() {
+        expect("\\text{hello $x + \\text{world $y$} + z$}").toParse();
+        expect("\\text{hello \\(x + \\text{world $y$} + z\\)}").toParse();
+        expect("\\text{hello $x + \\text{world \\(y\\)} + z$}").toParse();
+        expect("\\text{hello \\(x + \\text{world \\(y\\)} + z\\)}").toParse();
+    });
+
+    it("should forbid \\( within math mode", function() {
+        expect("\\(").toNotParse();
+        expect("\\text{$\\(x\\)$}").toNotParse();
+    });
+
+    it("should forbid $ within math mode", function() {
+        expect("$x$").toNotParse();
+        expect("\\text{\\($x$\\)}").toNotParse();
+    });
+
+    it("should detect unbalanced \\)", function() {
+        expect("\\)").toNotParse();
+        expect("\\text{\\)}").toNotParse();
+    });
+
+    it("should detect unbalanced $", function() {
+        expect("$").toNotParse();
+        expect("\\text{$}").toNotParse();
+    });
+
+    it("should not mix $ and \\(..\\)", function() {
+        expect("$x\\)").toNotParse();
+        expect("\\(x$").toNotParse();
```

Comment:
Shouldn't be these two in the text mode like:
```js
        expect("\\text{$x\\)}").toNotParse();
        expect("\\text{\\(x$}").toNotParse();
```
?

Reply:
Yes, thanks for catching this!  (Otherwise we were just testing use of `$`/`\(` outside math mode again.)  Fixed.